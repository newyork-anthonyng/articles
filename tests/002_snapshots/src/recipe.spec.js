import reducer from './recipe';

describe('Without snapshots ðŸ˜¥', () => {
  it('should return initial state', () => {
    const expected = {
      ingredients: [],
      name: '',
    };

    expect(reducer(undefined, {})).toEqual(expected);
  });

  it('should handle ADD_INGREDIENT', () => {
    const action = {
      type: 'ADD_INGREDIENT',
      ingredient: 'Papaya',
    };
    const expected = {
      ingredients: ['Papaya'],
      name: '',
    };

    expect(reducer(undefined, action)).toEqual(expected);
  });

  it('should handle UPDATE_NAME', () => {
    const action = {
      type: 'UPDATE_NAME',
      name: 'Salad',
    };
    const expected = {
      ingredients: [],
      name: 'Salad',
    };

    expect(reducer(undefined, action)).toEqual(expected);
  });
});

describe('With snapshots ðŸŽ‰', () => {
  it('should return initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle ADD_INGREDIENT', () => {
    const action = {
      type: 'ADD_INGREDIENT',
      ingredient: 'Papaya',
    };

    expect(reducer(undefined, action)).toMatchSnapshot();
  });

  it('should handle UPDATE_NAME', () => {
    const action = {
      type: 'UPDATE_NAME',
      name: 'Salad',
    };

    expect(reducer(undefined, action)).toMatchSnapshot();
  });
});
