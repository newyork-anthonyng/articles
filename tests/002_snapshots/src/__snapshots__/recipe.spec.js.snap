// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With snapshots ðŸŽ‰ should handle ADD_INGREDIENT 1`] = `
Object {
  "ingredients": Array [
    "Papaya",
  ],
  "name": "",
}
`;

exports[`With snapshots ðŸŽ‰ should handle UPDATE_NAME 1`] = `
Object {
  "ingredients": Array [],
  "name": "Salad",
}
`;

exports[`With snapshots ðŸŽ‰ should return initial state 1`] = `
Object {
  "ingredients": Array [],
  "name": "",
}
`;
